### PR TITLE
CloseButton: Remove enzyme and convert to RTL

### PR DESCRIPTION
**Pull request by kieftrav**
## Package
lib-react-components

## Describe your changes
Converted tests for the CloseButton from Enzyme to RTL. 

## How to Review
[CloseButton Story](http://localhost:9001/?path=/story/components-closebutton--default)

## General
- [ ] Unit Tests are passing locally and on Github
- [ ] Storybook renders as expected
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-react-components/src/CloseButton/CloseButton.spec.js [hunk 1]
```diff
@@ -1,33 +1,23 @@
...
 
-describe('<CloseButton />', function () {
-  let wrapper
-  before(function () {
-    wrapper = shallow(<CloseButton closeFn={sinon.spy()} />)
+describe('Component > CloseButton', function () {
+  it('renders the default button', function () {
+    const DefaultStory = composeStory(Default, Meta)
+    render(<DefaultStory />)
+    expect(screen.getByRole('button', { disabled: false })).to.be.ok()
```
Comment (shaunanoordin) [inline packages/lib-react-components/src/CloseButton/CloseButton.spec.js:9]: ⚠️ Ahhah, we've got a small problem here - `disabled: false/true` isn't actually a proper option for getRole()

Check out the `options?` listed in [RTL's ByRole documentation](https://testing-library.com/docs/queries/byrole) -
there's `hidden`, `name`, `description`, etc, but you can't select by disabled state.

You can test this yourself by changing `disabled: false` to `disabled: true`, and the `expect()` will _still_ say, yep, that's definitely `ok()`. This means that getByRole() is ignoring the invalid "disabled" option.

Comment (shaunanoordin) [reply packages/lib-react-components/src/CloseButton/CloseButton.spec.js:9]: One option is to do check the `disabled` attribute separately.

For example, we can replace...
```
expect(screen.getByRole('button', { disabled: false })).to.be.ok()
```

With...
```
const button = screen.getByRole('button')
expect(button).to.be.ok()
expect(button.hasAttribute('disabled')).to.be.false()
```

File: packages/lib-react-components/src/CloseButton/CloseButton.spec.js [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-  describe('with a color prop', function () {
-    it('should set the icon colour', function () {
-      const colouredButton = shallow(<CloseButton color='neutral-6' closeFn={sinon.spy()} />)
-      const icon = colouredButton.prop('icon')
-      expect(icon.props.color).to.equal('neutral-6')
-    })
+  it('renders the disabled button', function () {
+    const DisabledStory = composeStory(Disabled, Meta)
+    render(<DisabledStory />)
+    expect(screen.getByRole('button', { disabled: true })).to.be.ok()
```
Comment (shaunanoordin) [inline packages/lib-react-components/src/CloseButton/CloseButton.spec.js:21]: Same comment as above: `disabled` is not a valid option for `getByRole()`

File: packages/lib-react-components/src/CloseButton/CloseButton.spec.js [hunk 3]
```diff
@@ -1,33 +1,23 @@
...
+    render(<DefaultStory />)
+    expect(screen.getByRole('button', { disabled: false })).to.be.ok()
   })
 
-  it('renders without crashing', function () {
-    expect(wrapper).to.be.ok()
+  it('renders the light button', function () {
+    const LightStory = composeStory(Light, Meta)
+    render(<LightStory />)
+    expect(screen.getByRole('button', { disabled: false })).to.be.ok()
```
Comment (shaunanoordin) [inline packages/lib-react-components/src/CloseButton/CloseButton.spec.js:15]: Same comment as above: `disabled` is not a valid option for `getByRole()`

**Review by shaunanoordin — CHANGES_REQUESTED** [review]
## PR Review

Package: `lib-react-components`

This PR makes updates to the CloseButton component's tests and Storybook story.

Code changes _mostly_ look good - the tests are testing basic rendering, and the storybook shows relevant use cases - but there's a small problem with how getByRole() is used.

### Status

Changes requested:
- [ ] [] remove `{ disabled: ... }` argument in `screen.getByRole('button', { disabled: false })`
  - Reason: "disabled" [isn't a valid option for getByRole](https://testing-library.com/docs/queries/byrole), and gives people reading the test a false assurance.
- [ ] [optional]: instead of trying to select a disabled/not-disabled button, use hasAttribute, e.g. `expect(myExampleButton.hasAttribute('disabled')).to.be.false()`, to determine if a button is disabled/not-disabled.

Other than that small thing, everything else looks good. 👌

File: packages/lib-react-components/src/CloseButton/CloseButton.spec.js [hunk 4]
```diff
@@ -1,33 +1,23 @@
...
-import { expect } from 'chai'
+import { render, screen } from '@testing-library/react'
+import { composeStory } from '@storybook/react'
+import Meta, { Default, Light, Disabled } from './CloseButton.stories.js'
 
-describe('<CloseButton />', function () {
-  let wrapper
-  before(function () {
-    wrapper = shallow(<CloseButton closeFn={sinon.spy()} />)
+describe('Component > CloseButton', function () {
```
[the remark this answers is not in the document]
Comment (kieftrav) [reply packages/lib-react-components/src/CloseButton/CloseButton.spec.js:5]: I'm not sure if there's any actual standard naming here. Probably something to talk about at some point

**Review by shaunanoordin — APPROVED** [review]
## PR Review (Update)

I'm sorry I've been sitting on this one's re-review for so long! I wasn't paying attention to the updates, and didn't realise that the feedback about changing `getByRole()`'s use of `disabled` has already been addressed. 👍 

LGTM, let's get this PR merged!

**Pull request merged**